python/lsst/daf/butler/core/serverModels.py

```diff
@@ -98,10 +104,11 @@ class QueryDatasetsModel(BaseModel):
 
     datasetType: ExpressionQueryParameter
     collections: Optional[ExpressionQueryParameter] = None
+    dimensions: Optional[List[str]] = None
     dataId: Optional[DataId] = None
     where: Optional[str] = None
     findFirst: bool = False
     components: Optional[bool] = None
     bind: Optional[DataId] = None
     check: bool = True
-    kwargs: Optional[DataId] = None
+    keyword_args: Optional[DataId] = None  # mypy refuses to allow kwargs in model
```

Comment:
This problem goes away if you do a little bit more standardization/normalization of the arguments on the client before or when making the model, such as calling `DataCoordinate.standardize` to merge `dataId` and `**kwargs` to generate something that's definitely a `DataCoordinate`.  Could go in a `classmethod` factory on the model class.

Reply:
I had forgotten that I can call `DataCoordinate.standardize` without a registry and so can fold those into the DataId before I send it.

Reply:
In almost every case sql registry calls `self.expandDataId` on the arguments. I think a `Registry.standardizeDataId()` method that folds in dataId+kwargs+defaults would work. It doesn't look like RegistryDefaults is amenable to pydantic because it (1) takes `**kwargs` and (2) needs registry to finish construction (making it incompatible with being immutable). I'd need to make a to_simple/from_simple and also allow `RegistryDefaults` to be constructed from a trusted dataId+collections+run rather than requiring a registry to check the dataId.